Написание тестов через расширение вне общих модулей #1057

Absolemus opened the pull request:
Добавлена возможность загружать тесты из модуля объекта обработки

реализация  #1057

сделанные изменения

- Добавлена возможность загружать тесты из модуля объекта метаданных
- Добавлена возможность загружать тесты из модуля менеджера метаданных

@vanessa-opensource/Collaborators - просьба прокомментировать и проверить

Absolemus committed:
Написание тестов через расширение вне общих модулей #1057

Добавлена возможность загружать тесты из модуля объекта обработки

Absolemus committed:
Тесты из объекта #1057

Добавлена возможность запуска тестов из расширения из модуля менеджера и модуля объекта

artbear commented on plugins/ЗагрузчикРасширения/ЗагрузчикРасширения.xml:
```diff
@@ -1,14 +1,14 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
-<MetaDataObject xmlns="http://v8.1c.ru/8.3/MDClasses" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:cmi="http://v8.1c.ru/8.2/managed-application/cmi" xmlns:ent="http://v8.1c.ru/8.1/data/enterprise" xmlns:lf="http://v8.1c.ru/8.2/managed-application/logform" xmlns:style="http://v8.1c.ru/8.1/data/ui/style" xmlns:sys="htt ...
-	<ExternalDataProcessor uuid="c7137d82-5f24-4db5-b7b5-eeb7e0b8a2c6">
+<MetaDataObject xmlns="http://v8.1c.ru/8.3/MDClasses" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:cmi="http://v8.1c.ru/8.2/managed-application/cmi" xmlns:ent="http://v8.1c.ru/8.1/data/enterprise" xmlns:lf="http://v8.1c.ru/8.2/managed-application/logform" xmlns:style="http://v8.1c.ru/8.1/data/ui/style" xmlns:sys="htt ...
```
предлагаю откатить этот хмл-файл, т.к. меняется режим совместимости

а он важен для работы продукта в старых версиях или режимах совместимости

artbear commented on plugins/ЗагрузчикРасширения/ЗагрузчикРасширения/Forms/Форма.xml:
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
-<MetaDataObject xmlns="http://v8.1c.ru/8.3/MDClasses" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:cmi="http://v8.1c.ru/8.2/managed-application/cmi" xmlns:ent="http://v8.1c.ru/8.1/data/enterprise" xmlns:lf="http://v8.1c.ru/8.2/managed-application/logform" xmlns:style="http://v8.1c.ru/8.1/data/ui/style" xmlns:sys="htt ...
+<MetaDataObject xmlns="http://v8.1c.ru/8.3/MDClasses" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:cmi="http://v8.1c.ru/8.2/managed-application/cmi" xmlns:ent="http://v8.1c.ru/8.1/data/enterprise" xmlns:lf="http://v8.1c.ru/8.2/managed-application/logform" xmlns:style="http://v8.1c.ru/8.1/data/ui/style" xmlns:sys="htt ...
```
и этот файл предлагаю откатить из-за увеличения версии платформы

artbear commented on plugins/ЗагрузчикРасширения/ЗагрузчикРасширения/Forms/Форма/Ext/Form.xml:
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
-<Form xmlns="http://v8.1c.ru/8.3/xcf/logform" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:dcscor="http://v8.1c.ru/8.1/data-composition-system/core" xmlns:dcsset="http://v8.1c.ru/8.1/data-composition-system/settings" xmlns:ent="http://v8.1c.ru/8.1/data/enterprise" xmlns:lf="http://v8.1c.ru/8.2/managed-application/log ...
+<Form xmlns="http://v8.1c.ru/8.3/xcf/logform" xmlns:app="http://v8.1c.ru/8.2/managed-application/core" xmlns:cfg="http://v8.1c.ru/8.1/data/enterprise/current-config" xmlns:dcscor="http://v8.1c.ru/8.1/data-composition-system/core" xmlns:dcssch="http://v8.1c.ru/8.1/data-composition-system/schema" xmlns:dcsset="http://v8.1c.ru/8.1/data-composition-system/settings" xmlns:ent="http://v8.1c.ru/8.1/data ...
```
и этот файл предлагаю откатить из-за увеличения версии платформы

Absolemus replied:
Откатил

artbear reviewed (changes requested):
есть замечания.

Absolemus committed:
Исправил формы

Absolemus committed:
Написание тестов через расширение вне общих модулей #1057

Написание тестов через расширение вне общих модулей #1057

Добавлена возможность загружать тесты из модуля объекта обработки

Тесты из объекта #1057

Добавлена возможность запуска тестов из расширения из модуля менеджера и модуля объекта

Absolemus committed:
Merge branch 'testFromObjects' of https://github.com/Absolemus/add into testFromObjects

Absolemus committed:
Еще откат изменений форм

Absolemus committed:
Убрал лишние пробелы

artbear commented:
@Absolemus Большущее спасибо!
